editor/common/src/EditorBuiltInIpc.ts

```diff
@@ -103,3 +104,136 @@ export interface BasicManipulationCommandIpc extends EditCommandIpc {
+export interface ElementGeometryResultProps {
+  /** The element's geometry stream graphic data in world coordinates */
+  graphic?: Uint8Array;
+  /** The element's geometry stream information */
+  geometry?: ElementGeometryInfo;
+  /** The element's range box (available as ElementGeometryInfo.bbox when returning geometry) */
+  range?: Range3dProps;
+  /** The element's category (available as ElementGeometryInfo.category when returning geometry) */
+  categoryId?: Id64String;
+  /** The result element id (different than source id when requesting insert instead of update) */
+  elementId?: Id64String;
+}
+
+/** @alpha */
+export enum SubEntityType {
+  /** A single bounded part of a surface */
+  Face = 0,
+  /** A single bounded part of a curve */
+  Edge = 1,
+  /** A single point */
+  Vertex = 2,
+}
+
+/** @alpha */
+export interface SubEntityProps {
+  /** Identifies the geometric primitive in the geometry stream that owns this sub-entity when there are more than one */
+  index?: number;
+  /** Identifies the type of sub-entity */
+  type: SubEntityType;
+  /** Identifies a face, edge, or vertex of the geometric primitive */
+  id: number;
+}
+
+/** @alpha */
+export interface SubEntityLocationProps {
+  /** The information to identify the sub-entity */
+  subEntity: SubEntityProps;
+  /** The face, edge, or vertex location in world coordinates from closest point or locate request */
+  point?: XYZProps;
+  /** The face normal in world coordinates of the identified location on sub-entity */
+  normal?: XYZProps;
+  /** The face or edge u parameter of identified location on sub-entity */
+  uParam?: number;
+  /** The face v parameter of identified location on sub-entity */
+  vParam?: number;
+}
+
+/** @alpha */
+export interface SubEntityAppearanceProps {
+  /** Category id for geometry. */
+  category: Id64String;
+  /** SubCategory id for geometry. */
+  subCategory?: Id64String;
+  /** Material id for geometry. */
+  material?: Id64String;
+  /** color of geometry. */
+  color?: ColorDefProps;
+  /** transparency of geometry. */
+  transparency?: number;
+  /** weight of geometry. */
+  weight?: number;
+}
+
+/** @alpha */
+export interface SubEntityGeometryProps {
+  /** The face, edge, or vertex graphic data in world coordinates */
+  graphic?: Uint8Array;
+  /** The face, edge, or vertex geometry in world coordinates */
+  geometry?: ElementGeometryDataEntry;
+  /** The face or edge range box for the sub-entity geometry */
+  range?: Range3dProps;
+  /** The appearance information for the sub-entity geometry */
+  appearance?: SubEntityAppearanceProps;
+}
+
+/** @alpha */
+export interface OffsetFacesProps {
+  /** The faces to offset */
+  faces: SubEntityProps | SubEntityProps[];
+  /** The offset to apply to all faces, or the offset for each face */
+  distances: number | number[];
+}
+
+/** @alpha */
+export interface SolidModelingCommandIpc extends EditCommandIpc {
+  clearElementGeometryCache: () => Promise<void>;

```

Comment:
all these interface methods can be declared:

```ts
clearElementGeometryCache(): Promise<void>;
createElementGeometryCache(id: Id64String, filter?: ElementGeometryCacheFilter): Promise<boolean>;
...
```
I discovered that after i started, but i think it looks much cleaner. Probably should change all the other IPC interfaces too.

Reply:
Ok, wasn't sure what was preferred...I found it done both ways...

Reply:
i didn't know you could use the `name(): Promise<>` syntax when i started. For one thing it makes it much easier to copy and paste into your implementation.